hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -144,13 +171,35 @@ public void stop() {
     if (executorService != null) {
       executorService.shutdown();
     }
-    if (hiveLocalDir != null) {
-      hiveLocalDir.delete();
-    }
     if (baseHandler != null) {
       baseHandler.shutdown();
     }
     METASTORE_THREADS_SHUTDOWN.invoke();
+    HMS_HANDLER_THREAD_LOCAL_CONF.remove();
+    HMS_HANDLER_THREAD_LOCAL_TXN.remove();
+    Object connPool = CONN_POOL.get();
+    CONN_POOL.set(null);
+    if (connPool instanceof Closeable) {
+      try {
+        ((Closeable) connPool).close();
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to close connection pool in TxnHandler", e);
+      }
+    }
+    OBJECT_STORE_PROP.set(null);
+    PersistenceManagerFactory pmf = OBJECT_STORE_PMF.get();
+    if (pmf != null) {
+      pmf.close();
+    }
+    if (hiveLocalDir != null && hiveLocalDir.exists()) {
+      try {
+        Path localDirPath = new Path(hiveLocalDir.getAbsolutePath());
+        FileSystem fs = Util.getFs(localDirPath, hiveConf);
+        Preconditions.checkState(fs.delete(localDirPath, true), "Failed to delete " + localDirPath);
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to delete local dir " + hiveLocalDir.getAbsolutePath(), e);
```

Comment:
Why not just allow the exception to be propagated?

Reply:
Do you mean to allow this method to throw `IOException`? Then the signature needs update and will probably incur more changes. Will do that if it's OK.

Reply:
I would vote for extending the method throw clause.

Reply:
Done